client/src/cordova/android/OutlineAndroidLib/outline/src/main/java/org/outline/vpn/VpnTunnelService.java

```diff
@@ -300,13 +303,26 @@ private synchronized PlatformError startTunnel(
       startNetworkConnectivityMonitor();
     }
 
+    this.outlineClient.setKeyId(config.id);
     // Start exchanging traffic between the local TUN device and the remote device.
     final ConnectOutlineTunnelResult result =
-            Tun2socks.connectOutlineTunnel(this.tunFd.getFd(), client, remoteUdpForwardingEnabled);
+            Tun2socks.connectOutlineTunnel(this.tunFd.getFd(), this.outlineClient, remoteUdpForwardingEnabled);
     if (result.getError() != null) {
       tearDownActiveTunnel();
       return result.getError();
     }
+    // Set the cookie file path for reporting
+    try {
+        File filesDir = this.getFilesDir();
+        if (filesDir != null) {
+            this.outlineClient.setCookieFilePath(filesDir.getAbsolutePath());
+            this.outlineClient.startReporting();
+        } else {
+            LOG.log(Level.SEVERE, "Cookies path is null");
+        }
```

Comment:
Let's move this so it's always called on session start.

```suggestion
        } else {
            LOG.log(Level.SEVERE, "Cookies path is null");
        }
        this.outlineClient.startSession();
```